### ISSUE TITLE
fichier: server-side rename, move is broken

### ISSUE THREAD

**Issue by disconnect5852** [rclone/rclone#7856]
I'm using 1fichier as backup also mounted with clone, and it have many limitations, and here is one that makes work rclone not so well, however this can be workarounded, but it would be nice if rclone would respect this.
As i see, on 1 fichier renaming, so server-side move only inside the same folder, but not between different folders, even if one is a subfolder. When i try to move a file from an 1fichier folder to a different one on the same server, error, happens:
`2024/05/19 21:55:18 ERROR : gopro/20240306.mp4: Dir.Rename error: couldn't rename file: No file has been renamed #312
2024/05/19 21:55:18 ERROR : IO error: couldn't rename file: No file has been renamed #312`, the error is appears in file manager too

For this the WA is to copy the file to the other folder, then delete the original, manually, which happens like this: rclone dowloads the file to the vfs cache, then re-uploads the file to the other folder. But 1fichier supports server-side copy, i usually sync a folder to another, on the same 1fichier account, and all files are copied server side.

So even if server side move is very limited, it would be nice to prepare rclone for these limitations, so do the usualy rename in the same folder, but on moving a file, do a server-side-copy, and delete the original automatically, thus making it transparent to the user.

`rclone mount hasher1fichier:/ /home/sanyika/cloudkripta/ --daemon --vfs-cache-mode minimal --vfs-fast-fingerprint --server-side-across-configs --hasher-auto-size 1P --log-file /home/sanyika/logs/cloudkripta.log --fast-list -v`

```
rclone v1.66.0
- os/version: linuxmint 21.3 (64 bit)
- os/kernel: 5.15.0-102-generic (x86_64)
- os/type: linux
- os/arch: amd64
- go/version: go1.22.1
- go/linking: static
- go/tags: none
```

**Comment by ncw** [comment]
@disconnect5852 can you make a way for us to reproduce this problem please, then we can investigate - thanks

**Comment by ncw** [comment]
@disconnect5852 you are correct - this was easy to reproduce!

Please give this a try

[v1.68.0-beta.8268.933622d13.fix-7856-fichier](https://beta.rclone.org/branch/fix-7856-fichier/v1.68.0-beta.8268.933622d13.fix-7856-fichier/) on branch [fix-7856-fichier](https://github.com/rclone/rclone/tree/fix-7856-fichier) (uploaded in 15-30 mins)

I think the main problem here was that the flood detection had broken (1fichier changed the error number) so the integration tests were failing because of that, so this error was drowned in the noise.

The above fixes the flood detection and the rename problem so hopefully all will be well.

**Comment by ncw** [comment]
Thanks for testing. I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org/) in 15-30 minutes and released in v1.68
